uxarray/grid/grid.py

```diff
@@ -45,26 +46,14 @@ class Grid:
 
     Parameters
     ----------
-    input_obj : xarray.Dataset, ndarray, list, tuple, required
-        Input ``xarray.Dataset`` or vertex coordinates that form faces.
-
-    Other Parameters
-    ----------------
-    gridspec: bool, optional
-        Specifies gridspec
-    islatlon : bool, optional
-        Specify if the grid is lat/lon based
-    isconcave: bool, optional
-        Specify if this grid has concave elements (internal checks for this are possible)
-    use_dual: bool, optional
-        Specify whether to use the primal (use_dual=False) or dual (use_dual=True) mesh if the file type is mpas
-
-    Raises
-    ------
-        RuntimeError
-            If specified file not found or recognized
+    grid_ds : xr.Dataset
+        ``xarray.Dataset`` encoded in the UGRID conventions
+
+    grid_spec : str, default="UGRID"
+        Original unstructrued grid format (i.e. UGRID, MPAS, etc.)
 
-    Examples
+    ugrid_mapping : dict, default=None
+        mapping of ugrid dimensions and variables to source dataset's conventions
```

Comment:
> Technically, this public constructor can be explicitly called by the user to construct a Grid object (without being called by from_dataset() etc) such as:

This is a valid concern, and it brings up a larger discussion regarding how our design is intended to be used by the user.

The main motivation in re-writing our constructor to use class methods is because in it's current state, our constructor for the `Grid` was a "catch all" approach", which quickly became cluttered and was starting to become difficult to maintain.

Our `ux.open_grid` already acts as a "catch all".

To address your point, yes the user can directly call the `Grid` constructor, however I don't see it as being an issue (however we do need to update the docstring to better reflect the following i'm going to mention)

#### Case 1: General Grid Construction
```Python
uxgrid = ux.open_grid(grid_path, other_arguments)
```
#### Case 2: Specialized Grid Construction
Instead of using the "catch-all" method, the user instead want's to directly construct a `Grid` using one of the provided class methods
```Python
uxgrid = Grid.from_dataset()

uxgrid = Grid.from_face_verticies()

uxgrid = Grid.delaunay()

uxgrid = Grid.from_structured()

# etc
```

#### Case 3: Direct Construction from UGRID
By using the class methods, we essentially have the following flow:
1. Class method is called with source data
2. Class method encodes the source data in the UGRID conventions
3. UGRID dataset and variables are passed into the `Grid` constructor and returned

This means that our default `Grid` constructor always assumes the inputs to be in the UGRID conventions, which is in line with our design. So now, instead of needing to handle every possible input, case, etc. in our default Constructor, specialized class methods can be called from our `ux.open_grid` depending the use case. 

```Python
uxgrid = Grid(ugrid_ds)
```

I do agree that some tweaking and modifications need to be done to the default values in here, but let me know what you think of this workflow.

Reply:
Yes, this workflow makes sense to me; all is good. 

> The main motivation in re-writing our constructor to use class methods is because in it's current state, our constructor for the Grid was a "catch all" approach", which quickly became cluttered and was starting to become difficult to maintain.

Great change to our code!